source/fundamentals/crud/write-operations/insert.txt

```diff
@@ -4,9 +4,188 @@ Insert a Document
+Before you can retrieve, update, or delete information in MongoDB, you
+first need to perform an **insert operation** to load data. An insert
+operation inserts one or more documents into a MongoDB collection
+using one of the following methods: ``insertOne()``, ``insertMany()`` and ``bulkWrite()``.
 
+The following sections focus on ``insertOne()`` and ``insertMany()``. For an
+example on how to use the ``bulkWrite()`` method, see the :doc:`Bulk
+Operations Usage Example </usage-examples/bulkWrite>`.
+
+A Note About ``_id``
+~~~~~~~~~~~~~~~~~~~~
+
+When inserting a document, MongoDB enforces one constraint on your
+documents by default, which is that each document *must* contain a unique ``_id``
+field.
+
+There are two ways to manage this field:
+
+- You can manage this field yourself, ensuring each value you use is unique.
+- You can let the driver automatically generate unique ``ObjectId`` values.
+
+Unless you provide strong guarantees for uniqueness, we recommend
+you let the driver automatically generate ``_id`` values.
+
+.. note::
+
+   Duplicate ``_id`` values violate unique index constraints, which
+   causes the driver to throw a ``WriteError``.
+ 
+For more information about the ``_id`` field, see the Server Manual Entry on
+:manual:`Unique Indexes </core/index-unique/>`.
+
+Insert a Single Document
+------------------------
+
+Use the ``InsertOne()`` method to insert a single document into a collection.
+
+Upon successful insertion, the method returns an
+``InsertOneResult`` instance that contains the ``_id`` of
+the new document.
+
+Example
+~~~~~~~
+
+The following example creates and inserts a document into the
+``favorite_books`` collection using the
+``insertOne()`` method:
+
+.. code-block:: go
+
+   coll := client.Database("myDB").Collection("favorite_books")
+   doc := bson.D{{"title", "Invisible Cities"}, {"author", "Italo Calvino"}, {"year_published", 1974}}
+
+   result, err := coll.InsertOne(context.TODO(), doc)
+
+   fmt.Printf("Inserted document with _id: %v\n", result.InsertedID)
+
+Your output should look something like this:
+
+.. code-block::
+   :copyable: false
+
+   Inserted document with _id: ObjectID("6112c7cacf7fb532a9c0077f")
+
+For more information on the types and functions mentioned in this
+section, see the following resources:
+
+- `InsertOne() API documentation <https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo@v1.7.0#Collection.InsertOne>`__
+- `InsertOneResult API documentation <https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo@v1.7.0#InsertOneResult>`__
+- Server Manual Entry on :manual:`insertOne() </reference/method/db.collection.insertOne/>`
+- Runnable :doc:`Insert a Document Usage Example </usage-examples/insertOne>`
+
+Insert Multiple Documents
+-------------------------
+
+Use the ``InsertMany()`` method to insert multiple documents into a
+collection. The ``InsertMany()`` method returns an ``InsertManyResult``
+instance that contains the ``_id`` fields of the inserted documents.
```

Comment:
Nit: Can you make the wording and formatting in both sections consistent -- they contain the same content, but are worded and formatted differently. Either one is fine, but let's stick to only using that one kind.